src/asl/synth.ts

```diff
@@ -1894,12 +1894,77 @@ export class ASL {
               return { value: !!argOutput.value, containsJsonPath: false };
             }
           });
+        } else if (ref === String) {
+          const [arg] = expr.args;
+          if (!arg) {
+            return {
+              // String()
+              value: "",
+              containsJsonPath: false,
+            };
+          }
+          return this.evalExprToJsonPathOrLiteral(arg.expr, (argOutput) => {
+            // String(var) or String({ key: var })
+            if (ASLGraph.isJsonPath(argOutput) || argOutput.containsJsonPath) {
+              const temp = this.newHeapVariable();
+              const jsonPathStates = this.normalizeOutputToJsonPath(argOutput);
+              const { jsonPath } = ASLGraph.isJsonPath(jsonPathStates)
+                ? jsonPathStates
+                : jsonPathStates.output;
+
+              return {
+                ...ASLGraph.joinSubStates(arg, jsonPathStates, {
+                  // if this was a literal, we know it was an object, not a string
+                  startState: ASLGraph.isLiteralValue(argOutput)
+                    ? "format"
+                    : "checkString",
+                  states: {
+                    // String("") => ""
+                    checkString: {
+                      Type: "Choice",
+                      Choices: [{ ...ASL.isString(jsonPath), Next: "assign" }],
+                      Default: "format",
+                    },
+                    format: {
+                      Type: "Pass",
+                      Parameters: {
+                        "str.$": `States.JsonToString(${jsonPath})`,
+                      },
+                      ResultPath: temp,
+                      Next: ASLGraph.DeferNext,
+                    },
+                    // since this state can only output a single json path value,
+                    // assign the input json path to the expected output path
+                    assign: {
+                      Type: "Pass",
+                      InputPath: jsonPath,
+                      ResultPath: `${temp}.str`,
+                      Next: ASLGraph.DeferNext,
+                    },
+                  },
+                })!,
+                output: {
+                  jsonPath: `${temp}.str`,
+                },
+              };
+            } else {
+              // String(1)
+              return {
+                value: String(argOutput.value),
+                containsJsonPath: false,
+              };
+            }
+          });
+        } else if (ref === Number) {
+          const [arg] = expr.args;
+          return this.evalToNumber(arg?.expr);
         }
       }
       throw new Error(
-        `call must be a service call or list .slice, .map, .forEach or .filter: ${toStateName(
-          expr
-        )}`
+        `call must be an integration call, list(.slice, .map, .forEach, .filter, or ` +
```

Comment:
typeo, `list(`

Reply:
I meant to do that, was trying to show `list(.slice, .map, ...)`